model_compression_toolkit/pytorch/back2framework/model_builder.py

```diff
@@ -121,23 +122,34 @@ class PytorchModelBuilder(torch.nn.Module):
     """
     def __init__(self, graph: Graph,
                  mode: ModelBuilderMode = ModelBuilderMode.QUANTIZED,
-                 append2output: List[Any] = None):
+                 append2output: List[Any] = None,
+                 fw_info: FrameworkInfo = DEFAULT_PYTORCH_INFO):
         """
         Construct a Pytorch model.
         Args:
             graph: Graph to build its corresponding Pytorch model.
             mode: Building mode. Read ModelBuilderMode description for more info.
             append2output: List of nodes or OutTensor objects.
+            fw_info: Framework information (e.g., mapping from layers to their attributes to quantize).
         """
         super(PytorchModelBuilder, self).__init__()
         self.graph = graph
         self.mode = mode
         self.node_sort = list(topological_sort(graph))
         self.nodes_dict = {}
         self.append2output = append2output
-        for n in self.node_sort:
-            if not isinstance(n, FunctionalNode):
-                self.add_module(n.name, node_builder(n))
+
+        if mode == ModelBuilderMode.MIXEDPRECISION:
+            for n in self.node_sort:
+                if not isinstance(n, FunctionalNode):
+                    if n.is_weights_quantization_enabled():
+                        self.add_module(n.name, PytorchMixedPrecisionWrapper(n, fw_info))
+                    else:
+                        self.add_module(n.name, node_builder(n))
+        else:
+            for n in self.node_sort:
```

Comment:
Let's use the get_configurable_nodes the graph has. This should be the source of information for which nodes we need to search bit-widths.

Reply:
You mean instead of using 
`self.sorted_nodes = list(topological_sort(graph))`?
or instead of checking the condition `if n.is_weights_quantization_enabled()`?